script.js

```diff
@@ -0,0 +1,336 @@
+        const question = quizQuestions[currentQuestionIndex];
+        displayQuestion(question, currentQuestionIndex + 1);
+      } else {
+        if (currentPlayer === 1) {
+          currentPlayer = 2;
+          currentQuestionIndex = 0;
+          displayNextQuestion();
+        } else {
+          displayFinalScores();
+        }
+      }
+      updateCurrentPlayerDisplay();
+    }
+
+    //Function to display a questions in the quiz
+    function displayQuestion(question, questionNumber) {
+      quizDisplay.innerHTML = '';
+
+      const listItem = document.createElement('li');
+      listItem.className = 'question-list-in-quiz-game';
+      const questionText = document.createElement('p');
+      questionText.className = 'question-text-in-quiz-game';
+      questionText.textContent = `Question ${questionNumber}: ${question.question}`;
+      listItem.appendChild(questionText);
+
+      const optionsList = document.createElement('ul');
+      optionsList.className = 'options-list-in-quiz-game';
+      question.options.forEach((option, i) => {
+        const optionItem = document.createElement('li');
+        optionItem.className = 'option-item-in-quiz-game';
+        const optionButton = document.createElement('button');
+        optionButton.className = 'option-button-in-quiz-game';
+        optionButton.textContent = `${option.text}`;
+        optionButton.addEventListener('click', function () {
+          handleAnswerClick(i, question.correctIndex, optionsList);
+        });
+        optionItem.appendChild(optionButton);
+        optionsList.appendChild(optionItem);
+      });
+      listItem.appendChild(optionsList);
+
+      quizDisplay.appendChild(listItem);
+    }
+
+ // Function to handle a user's answer click
+ function handleAnswerClick(selectedIndex, correctIndex, optionsList) {
+    const isCorrect = selectedIndex === correctIndex;
+  
+    // Highlight the selected option
+    const selectedOptionItem = optionsList.children[selectedIndex];
+    const optionButton = selectedOptionItem.querySelector('.option-button-in-quiz-game');
+    
+    // Change the button color based on correctness
+    optionButton.style.backgroundColor = isCorrect ? 'green' : 'red';
+  
+    // Highlight the correct answer
+    const correctOptionItem = optionsList.children[correctIndex];
+    const correctOptionButton = correctOptionItem.querySelector('.option-button-in-quiz-game');
+    correctOptionButton.style.backgroundColor = 'green';
+  
+    if (isCorrect) {
+      currentPlayer === 1 ? player1ScoreDisplay.textContent++ : player2ScoreDisplay.textContent++;
+    }
+
+      setTimeout(() => {
+        // Reset styles for the next question
+        selectedOptionItem.classList.remove('correct', 'incorrect');
+        correctOptionItem.classList.remove('correct', 'incorrect');
+        currentQuestionIndex++;
+        displayNextQuestion();
+        updateCurrentPlayerDisplay();
+      }, 2000);
```

Comment:
2 seconds feels little bit too long, I would try shorter time, something between 500ms and 1500ms. Right now I think you wait too long and it feels like something is broken in the interaction.